test/Constraints/diagnostics.swift

```diff
@@ -1031,7 +1031,6 @@ takesClosure() { ambiguousCall() } // expected-error {{ambiguous use of 'ambiguo
 class SR_4692_a {
   private static func foo(x: Int, y: Bool) {
     self.bar(x: x)
-    // expected-error@-1 {{instance member 'bar' cannot be used on type 'SR_4692_a'}}
```

Comment:
What did be lose errors in this and `SR_4692_b` cases?

Reply:
I am not sure why we lost the diagnostic here, I'll look into this. I am a bit surprised that we've regressed further here as nothing's changed in my code since last time. Perhaps it's due to the recent changes to diagnostics code.